lib/internal/event_target.js

```diff
@@ -75,9 +75,15 @@ class Event {
     this[kDefaultPrevented] = false;
     this[kTimestamp] = lazyNow();
     this[kPropagationStopped] = false;
+    this[kTrustEvent] = false;
+    if (options != null && options[kTrustEvent]) {
+      this[kTrustEvent] = options[kTrustEvent];
+    }
     // isTrusted is special (LegacyUnforgeable)
     ObjectDefineProperty(this, 'isTrusted', {
-      get: isTrusted,
+      get() {
+        return this[kTrustEvent];
+      },
```

Comment:
We should avoid this:

- It tanks perf, see https://github.com/nodejs/node/pull/34459 / https://twitter.com/tverwaes/status/1285497796540473344
- If we make `kTrustEvent` a symbol property, it’s not unforgeable anymore, which means we might as well stop using an own property here

Reply:
@addaleax given these conflicting requirements (unforgeable and the fact we need some events to be trusted internally) - what should we do?

Regarding performance - I am not sure the performance of isTrusted for EventTarget is that important (is it?)

Reply:
@addaleax hmm, actually I can capture it from the closure - let me try and let me know what you think - though it will likely tank performance just as bad.

Reply:
>  I am not sure the performance of isTrusted for EventTarget is that important (is it?)

It’s not the performance of `isTrusted`, it’s the performance of `new Event()` that matters here :)

Reply:
@addaleax please take a look. I think what I ended up doing shouldn't tank performance or introduce a closure. Should I run benchmarks?

(Also thanks, I didn't know this tanks perf 🙏 )

Reply:
> Should I run benchmarks?

I think that would be helpful, yes :) And maybe also update the benchmarks to account for situations in which sometimes `new Event()` creates a trusted event and sometimes not :)